perf_test.go

```diff
@@ -0,0 +1,46 @@
+package epub
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func BenchmarkAddImage_http(b *testing.B) {
```

Comment:
very cool! I didn't realize go had built-in benchmarking (https://pkg.go.dev/testing#hdr-Benchmarks)

Reply:
Yes, and it allows analysis easily.
Try for example:
```
go test -bench=http -benchmem -memprofile memprofile.out -cpuprofile profile.out
go tool pprof -http=:8081 memprofile.out
```

This showed me that testing local file is cheap compared to testing http. Therefore it is more efficient to test local before.